src/pagination/mod.rs

```diff
@@ -136,23 +135,45 @@ fn interpret_permalink(
     page_num: usize,
     index: Option<&liquid::value::Value>,
 ) -> Result<String> {
-    Ok(if page_num == 1 {
-        if let Some(index) = index {
-            let include: &str = config.include.into();
-            format!("{}/{}", include, index_to_string(&index))
-        } else {
-            doc.url_path.clone()
-        }
+    let mut attributes = document::permalink_attributes(&doc.front, &doc.file_path);
+    let permalink = permalink::explode_permalink(&config.front_permalink, &attributes)?;
+    let permalink_path = std::path::Path::new(&permalink);
```

Comment:
Not sure how I feel about using `std::path` for manipulating a URI path

Reply:
Is there another tool in the std lib to test for extension?

Reply:
I'd either handlroll a solution (like we've done with other url stuff) or use [`Url::path_segments`](https://docs.rs/url/1.7.2/url/struct.Url.html#method.path_segments) to at least get the last segment and then check for the extension within that.